lib/spack/spack/main.py

```diff
@@ -405,9 +398,6 @@ def make_argument_parser(**kwargs):
         '-d', '--debug', action='count', default=0,
         help="write out debug messages "
              "(more d's for more verbosity: -d, -dd, -ddd, etc.)")
-    parser.add_argument(
```

Comment:
This flag is already in 0.17, maybe just turn it into a no-op and remove it in 0.19-dev.